atlp-rwanda/Phantom-Backend-Elites
# User roles and permissions

## What does this PR do?
- Adds CRUD for user `Roles`
- Adds CRUD for user Role's `Permissions`
- Makes it possible for the in-charge person to `assign Permissions to Roles`
- Allows `users` to be `assigned Roles`
## Description:
This is a new idea that I have came across, which allows the **administrator**
to `manage roles and permissions` as well as assigning different roles to new and existing users,
**assigning different permissions to different roles** and more. To keep this PR simple, 
other logics will be implemented in another PR to come.

## How should this be manually tested?
​
- clone the repository,then run `npm i` to install all depandencies.
- start the server with `npm start`
- Go to `http://localhost:3000/api/v1/roles` to see a list of roles.  *You can even append an id to manipulate a role*
- Go to `http://localhost:3000/api/v1/permissions` to see a list of permissions. *You may an id to manipulate a permission*
- Go to `http://localhost:3000/api/v1/assign_perm` to see a list of roles.  *You may use this to assign a permission to a role*
- Go to `http://localhost:3000/api/v1/assign_role` to see a list of permissions. *You may use this to assign a role to a user*
